portfolio/src/main/java/com/google/sps/data/CommentListSingleton.java

```diff
@@ -0,0 +1,53 @@
+package com.google.sps.data;
+
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Map;
+import java.util.HashMap;
+
+import com.google.sps.dataExceptions.CommentExistingId;
+import com.google.sps.dataExceptions.CommentNotFound;
+
+public class CommentListSingleton {
+    private static CommentListSingleton instance;
+
+    public static CommentListSingleton getInstance() {
+        if(instance == null) {
+            instance = new CommentListSingleton();
+        }
+
+        return instance;
+    }
+
+    private Map <Integer, Comment> commentsContainer;
+
+    private CommentListSingleton() {
+        this.commentsContainer = new HashMap<Integer, Comment>(); // hash table-like implementation (allows insertions, deletions and search by id in O(1))
```

Comment:
HasMap not a good fit in your scenario. Web server is a multithreaded system and this container might be used while processing multiple simultaneous requests and it will lead to ConcurrentModificationException.

Reply:
So what's better in such a case? Is a List a good option?

Reply:
package java.util.concurrent contains collections of various flavor that can be safely used in multithreaded environment. No need to take an action at this cl, as current code is temporary solution.

Reply:
As agreed on the call, we will leave this now as it is. However, I researched about it and learned some new stuff! Also, this will not be an issue afterwards, as all the comments will be stored on the datastore!